cypress/commons/utils/apiUtils.js

```diff
@@ -286,6 +294,71 @@ const interceptGetDatasets = () => {
   return alias;
 };
 
+// Parameters:
+//   - options: dict with properties:
+//     - id (optional): id of the dataset to create
+//     - validateRequest (optional): a function, taking the request object as argument, that can be used to perform
+//       cypress checks on the content of the intercepted query
+const interceptCreateDataset = (options) => {
+  const alias = forgeAlias('reqCreateDataset');
+  cy.intercept({ method: 'POST', url: API_REGEX.DATASETS, times: 1 }, (req) => {
+    if (options?.validateRequest) options?.validateRequest(req);
+    if (stub.isEnabledFor('CREATE_DATASET')) {
+      const dataset = {
+        ...DEFAULT_DATASET,
+        ...req.body,
+        id: options.id ?? `d-stbd${utils.randomStr(4)}`,
+      };
+
+      if (stub.isEnabledFor('GET_DATASETS')) {
+        stub.addDataset(dataset);
+      }
+      req.reply(dataset);
+    } else if (stub.isEnabledFor('GET_DATASETS')) {
+      req.continue((res) => stub.addDataset(res.body));
+    }
+  }).as(alias);
+  return alias;
+};
+
+// Parameters:
+//   - options: dict with properties:
+//     - validateRequest (optional): a function, taking the request object as argument, that can be used to perform
+//       cypress checks on the content of the intercepted query
+//     - customDatasetPatch (optional): data to set in the request body, you can use this option to replace the original
+//       content of the query
+const interceptUpdateDataset = (options) => {
+  const alias = forgeAlias('reqUpdateDataset');
+  cy.intercept({ method: 'PATCH', url: API_REGEX.DATASET, times: 1 }, (req) => {
+    if (options?.validateRequest) options?.validateRequest(req);
+    const datasetPatch = {
+      ...req.body,
+      ...options?.customDatasetPatch,
+    };
+    // const datasetId = options?.datasetId ?? req.url.match(API_REGEX.DATASET)[1];
+    const datasetId = req.url.match(API_REGEX.DATASET)[1];
+    if (stub.isEnabledFor('GET_DATASETS')) stub.patchDataset(datasetId, datasetPatch);
+    if (stub.isEnabledFor('UPDATE_DATASET')) req.reply(datasetPatch);
+  }).as(alias);
+  return alias;
+};
+
+// Parameters:
+//   - fileName: name (or path in Storage) of the uploaded file
+//   - fileContent: content of the uploaded file
+const interceptUploadWorkspaceFile = (fileName, fileContent) => {
+  const alias = forgeAlias('reqUploadWorkspaceFile');
+  cy.intercept({ method: 'POST', url: API_REGEX.FILE_UPLOAD, times: 1 }, (req) => {
+    if (stub.isEnabledFor('CREATE_DATASET')) {
+      stub.addWorkspaceFile(fileName, fileContent);
+      req.reply(fileName);
+    } else if (stub.isEnabledFor('GET_DATASETS')) {
+      req.continue((res) => stub.addWorkspaceFile(fileName, fileContent));
```

Comment:
`res` argument isn't used, it is mandatory for continue function? :thinking:

Reply:
We can probably remove it from the arguments :+1: